data-prepper-core/src/integrationTest/java/org/opensearch/dataprepper/integration/ProcessorSwapPipelineIT.java

```diff
@@ -0,0 +1,141 @@
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.empty;
+import static org.opensearch.dataprepper.test.framework.DataPrepperTestRunner.BASE_PATH;
+
+class ProcessorSwapPipelineIT {
+    private static final Logger LOG = LoggerFactory.getLogger(ProcessorSwapPipelineIT.class);
+    private static final String IN_MEMORY_IDENTIFIER = "ProcessorSwapPipelineIT";
+    private static final String PIPELINE_CONFIGURATION_FOLDER_UNDER_TEST = "processor-swap";
+    private static final String PIPELINE_NAME_IN_YAML = "processor-pipeline";
+    private DataPrepperTestRunner dataPrepperTestRunner;
+    private InMemorySourceAccessor inMemorySourceAccessor;
+    private InMemorySinkAccessor inMemorySinkAccessor;
+    private PluginFactory pluginFactory;
+
+    @BeforeEach
+    void setUp() {
+        dataPrepperTestRunner = DataPrepperTestRunner.builder()
+                .withPipelinesDirectoryOrFile(PIPELINE_CONFIGURATION_FOLDER_UNDER_TEST + "/source")
+                .build();
+
+        inMemorySourceAccessor = dataPrepperTestRunner.getInMemorySourceAccessor();
+        inMemorySinkAccessor = dataPrepperTestRunner.getInMemorySinkAccessor();
+        pluginFactory = dataPrepperTestRunner.getPluginFactory();
+        dataPrepperTestRunner.start();
+        LOG.info("Started test runner.");
+    }
+
+    @AfterEach
+    void tearDown() {
+        LOG.info("Test tear down. Stop the test runner.");
+        dataPrepperTestRunner.stop();
+    }
+
+    @Test
+    void run_with_single_record() {
+        final String messageValue = UUID.randomUUID().toString();
+        final Event event = JacksonEvent.fromMessage(messageValue);
+        final Record<Event> eventRecord = new Record<>(event);
+
+        LOG.info("Submitting a single record.");
+        inMemorySourceAccessor.submit(IN_MEMORY_IDENTIFIER, Collections.singletonList(eventRecord));
+
+        await().atMost(400, TimeUnit.MILLISECONDS)
+                .untilAsserted(() -> assertThat(inMemorySinkAccessor.get(IN_MEMORY_IDENTIFIER), not(empty())));
+
+        final List<Record<Event>> records = inMemorySinkAccessor.get(IN_MEMORY_IDENTIFIER);
+
+        assertThat(records.size(), equalTo(1));
+
+        assertThat(records.get(0), notNullValue());
+        assertThat(records.get(0).getData(), notNullValue());
+        assertThat(records.get(0).getData().get("message", String.class), equalTo(messageValue));
+        assertThat(records.get(0).getData().get("test1", String.class), equalTo("knownPrefix10"));
+        assertThat(records.get(0).getData().get("test1_copy_original", String.class), equalTo("knownPrefix10"));
+
+        // Dynamically swap the pipeline processors
+        LOG.info("Swapping the pipeline processors");
+        List<Processor> targetPipelineProcessors = getTargetPipelineProcessors();
+        dataPrepperTestRunner.swapProcessors(PIPELINE_NAME_IN_YAML, targetPipelineProcessors);
```

Comment:
I think it would be need to use the HTTP endpoint itself. You can another integration test that exercises it here.

https://github.com/opensearch-project/data-prepper/blob/179e3dfd949031ec8faeb289c69d754beb5ee455/data-prepper-core/src/integrationTest/java/org/opensearch/dataprepper/integration/CoreHttpServerIT.java#L46-L53